spec/lib/ruby_smb/smb2/packet/error_packet_spec.rb

```diff
@@ -35,38 +35,17 @@
     it 'should be a 32-bit unsigned integer' do
       expect(packet.byte_count).to be_a BinData::Uint32le
     end
-
-    it 'should be the number of bytes in #error_data' do
-      str = 'testing'
-      packet.error_data = str
-      expect(packet.byte_count).to eq str.size
-    end
-
-    it 'should be 0 when #error_data is 1-byte long' do
-      packet.error_data = "\x00"
-      expect(packet.byte_count).to eq 0
-    end
   end
 
   describe '#error_data' do
     it 'should be a String' do
       expect(packet.error_data).to be_a BinData::String
     end
 
-    it 'should be \x00 by default' do
-      expect(packet.error_data).to eq "\x00"
-    end
-
-    it 'should read 1 byte when #byte_count is 0' do
-      packet.error_data = 'test'
-      packet.byte_count = 0
-      expect(described_class.read(packet.to_binary_s).error_data.size).to eq 1
-    end
-
-    it 'should read #byte_count bytes when #byte_count is not 0' do
+    it 'should read #byte_count bytes' do
       packet.error_data = 'test'
-      packet.byte_count = 4
-      expect(described_class.read(packet.to_binary_s).error_data.size).to eq 4
+      packet.byte_count = 3
+      expect(described_class.read(packet.to_binary_s).error_data.size).to eq 3
```

Comment:
The `error_data` that is provided is the same as before but the size was updated to `3`.
Is there a limitation to the data size?

Reply:
The `byte_count` limits the number of byte to be assigned to `error_data` when reading a raw packet (thanks to Bindata `:read_length` parameter). This example checks if only 3 bytes are assigned when reading a raw packet with `byte_count` set to 3, even if `error_data` string is 4-bytes long .

Reply:
That makes sense. Thanks.